pkg/garden/app/treaty.hoon

```diff
@@ -298,9 +302,17 @@
   ++  watching  (~(has by wex.bowl) [path dock])
   ++  safe-watch  `(unit card)`?:(|(watching =(our.bowl ship)) ~ `watch)
   ++  leave  (leave:pass dock)
+  ++  gone   
+    ^-  (list card)
+    :~  (fact:io (treaty-update:cg %del ship desk) /treaties ~)
+        (kick-only:io our.bowl path ~)
+    ==
```

Comment:
These arms feel like they might be safer to use if they do both the state modification and effects at once (ie `^-  (quip card treaty)` or w/e), but that might be out of scope for this pr.